components/expense/expense_api.go

```diff
@@ -0,0 +1,99 @@
+package expense
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/marian0/expenses-go-api/common"
+)
+
+//API -
+type API struct {
+	eService    *Service
+}
+
+//NewAPI -
+func NewAPI(expenseService *Service) *API {
+
+
+	return &API{
+		eService:	expenseService,
+	}
+}
+
+//FindAll -
+func (eAPI *API) FindAll(c *gin.Context) {
+	expenses := eAPI.eService.FindAll()
+
+	common.RespondSuccess(c, http.StatusOK, gin.H{"expenses": ToExpenseDTOs(expenses)})
+}
+
+//FindByID -
+func (eAPI *API) FindByID(c *gin.Context) {
+	id, _ := strconv.Atoi(c.Param("id"))
+	expense := eAPI.eService.FindByID(uint(id))
+
+	common.RespondSuccess(c, http.StatusOK, gin.H{"expense": ToExpenseDTO(&expense)})
+}
+
+//Create -
+func (eAPI *API) Create(c *gin.Context) {
+	//Logged User
+	// uuid, _ := c.Get(common.CONST_UUID_IDENTIFIER)
+
+	var expenseDTO CreateDTO
+
+	err := c.ShouldBindJSON(&expenseDTO)
+	if err != nil {
+		common.RespondError(c, http.StatusBadRequest, err.Error(), "There was a problem trying to create a Expense")
+		return
+	}
+
+	createdExpense := eAPI.eService.Create(&Expense{
```

Comment:
@zgoldy in the future we should check for errors here, what if expense creation fails ? We should return an error. Please, add at least a comment like so:

```
// @todo: Validate expense creation and handle errors 
```